src/lib/agent/AgentMessage.ts

```diff
@@ -0,0 +1,42 @@
+import { Expose } from 'class-transformer';
+import { Matches } from 'class-validator';
+import uuid from 'uuid/v4';
+
+import { Constructor, Compose } from '../utils/mixins';
+import { ThreadDecorated } from '../decorators/thread/ThreadDecoratorExtension';
+import { L10nDecorated } from '../decorators/l10n/L10nDecoratorExtension';
```

Comment:
It seems like we have a dependency on `decorators/l10n/L10nDecoratorExtension` and there we have a dependency on  `AgentMessage`

Reply:
We should separate `BaseMessage` from `AgentMessage`. That will solve this circular dependency.